includes/class-freemius.php

```diff
@@ -20992,23 +20992,43 @@ private function _sync_plugin_license(
-                                            'wp.freemius.com'
-                                        ) ) ) . '</b>'
-                                    ) . '<br> ' . $this->get_text_inline( 'Error received from the server:', 'server-error-message' ) . var_export( $result->error, true ),
-                                    $this->get_text_x_inline( 'Oops', 'exclamation', 'oops' ) . '...',
-                                    'error',
-                                    $background,
-                                    'api_blocked'
-                                );
-                            }
+	                        $add_notice = false;
+	                        if ( $background ) {
+		                        $counter = (int) get_transient( '_fs_api_connection_retry_counter' );
+
+                                // We only want to show the notice after 3 consecutive failures.
+		                        if ( $counter >= 3 ) {
+			                        $add_notice = true;
+			                        $counter    = 0;
+		                        }
+
+                                // Generally, background sync happens once a day. So, setting expiration for a week time should be enough.
+		                        set_transient( '_fs_api_connection_retry_counter', $counter + 1, WP_FS__TIME_WEEK_IN_SEC );
+	                        }
+
+	                        // Add notice instantly for not-background sync and only after 3 failed attempts for background sync.
+	                        if ( ( ! $background || $add_notice ) &&
+	                             ! self::$_global_admin_notices->has_sticky( 'api_blocked' )
+	                        ) {
+		                        self::$_global_admin_notices->add(
+			                        sprintf(
+				                        $this->get_text_inline( 'Your server is blocking the access to Freemius\' API, which is crucial for %1$s synchronization. Please contact your host to whitelist %2$s', 'server-blocking-access' ),
+				                        $this->get_plugin_name(),
+				                        '<b>' . implode( ', ', $this->apply_filters( 'api_domains', array(
+					                        'api.freemius.com',
+					                        'wp.freemius.com'
+				                        ) ) ) . '</b>'
+			                        ) . '<br> ' . $this->get_text_inline( 'Error received from the server:', 'server-error-message' ) . var_export( $result->error, true ),
+			                        $this->get_text_x_inline( 'Oops', 'exclamation', 'oops' ) . '...',
+			                        'error',
+			                        $background,
+			                        'api_blocked'
+		                        );
+	                        }
                         } else {
+                            // API is working now. Delete the transient and start afresh.
+                            delete_transient('_fs_api_connection_retry_counter');

```

Comment:
@msladu this part of the logic is executed when connectivity works but authentication params are wrong (which is rare and for edge cases). Please revise the method to position the transient deletion in a better place.

Reply:
@vovafeldman Looking into it.